spec/controllers/contributions_controller.spec.rb

```diff
@@ -0,0 +1,146 @@
+    end
+
+    it "should render the form elements" do
+      expect(page).to have_field('contribution[user_id]')
+      expect(page).to have_field('contribution[company_id]')
+      expect(page).to have_field('contribution[repository_id]')
+      expect(page).to have_field('contribution[link]')
+    end
+  end
+
+  describe "POST create" do
+    let(:user) { create(:user) }
+    let(:company) { create(:company) }
+    let(:repository) { create(:repository) }
+
+    let(:contribution_params) {
+      {
+        "link"=>"https://github.com/Codeminer", 
+        "user_id"=>"#{user.id}", 
+        "company_id"=>"#{company.id}",
+        "repository_id" => "#{repository.id}"
+      }
+    }
+
+
+    context "with valid params" do
+      it "creates a new Contribution" do
+        
+        expect {
+          post :create, params: { contribution: contribution_params }
+        }.to change(Contribution, :count).by(1)
+      end
+
+      it "assigns a newly created contribution as @contribution" do
+        post :create, params: { contribution: contribution_params }
+        expect(assigns(:contribution)).to be_a(Contribution)
+        expect(assigns(:contribution)).to be_persisted
+      end
+
+      it "redirects to the created contribution" do
+        post :create, params: { contribution: contribution_params }
+
+        expect(response).to have_http_status(:redirect)
+        expect(response).to redirect_to(admin_contribution_path(Contribution.last))
+      end
+
+      it 'should create the contribution' do
+        post :create, params: { contribution: contribution_params }
+        contribution = Contribution.last
+
+        expect(contribution.link).to eq(contribution_params['link'])
+        expect(contribution.user_id).to  eq(user.id)
+        expect(contribution.company_id).to      eq(company.id)
+        expect(contribution.repository_id).to      eq(repository.id)
+      end
+    end
+
+    context "with invalid params" do
+      it 'invalid_attributes return http success' do
+        post :create, params: { contribution: invalid_attributes }
+        expect(response).to have_http_status(:success)
+      end
+
+      it "assigns a newly created but unsaved contribution as @contribution" do
+        post :create, params: { contribution: invalid_attributes }
+        expect(assigns(:contribution)).to be_a_new(Contribution)
+      end
+
+      it 'invalid_attributes do not create a Contribution' do
+        expect do
+          post :create, params: { person: invalid_attributes }
+        end.not_to change(Contribution, :count)
+      end
+    end
+
+  end
+
+  describe "GET show" do
+    let(:user) { create(:user) }
+    let(:company) { create(:company) }
+    let(:repository) { create(:repository) }
+    let(:contribution_to_show) { create(:contribution, company_id: company.id, user_id: user.id, repository_id: repository.id, link: contribution[:link]) }
+
+
+    before do
+      get :show, params: { id: contribution_to_show.id }
```

Comment:
please add the exercise part of the test in the test, use before  only for setup

Reply:
The idea is put the statement `get :show, params: { id: contribution_to_show.id }` in every `it` case?

Reply:
yes